src/screen/subIdentities/RegisterSubIdentitiesScreen.tsx

```diff
@@ -0,0 +1,182 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {Alert, FlatList, StyleSheet, View} from 'react-native';
+import {Modalize} from 'react-native-modalize';
+import {useQueryClient} from 'react-query';
+import Identicon from '@polkadot/reactnative-identicon';
+import {NavigationProp, RouteProp} from '@react-navigation/native';
+import {Button, Card, Divider, IconProps, Layout, ListItem, Text, TopNavigationAction} from '@ui-kitten/components';
+import Icon from 'presentational/Icon';
+import ModalTitle from 'presentational/ModalTitle';
+import SafeView, {noTopEdges} from 'presentational/SafeView';
+import {useAccountIdentityInfo} from 'src/api/hooks/useAccountIdentityInfo';
+import {useApiTx} from 'src/api/hooks/useApiTx';
+import {SubIdentity, useSubIdentities} from 'src/api/hooks/useSubIdentities';
+import {DashboardStackParamList} from 'src/navigation/navigation';
+import {registerSubIdentitiesScreen} from 'src/navigation/routeKeys';
+import globalStyles, {monofontFamily} from 'src/styles';
+import {AddSubIdentity} from './AddSubIdentity';
+
+type ScreenProps = {
+  navigation: NavigationProp<DashboardStackParamList>;
+  route: RouteProp<DashboardStackParamList, typeof registerSubIdentitiesScreen>;
+};
+
+export function RegisterSubIdentitiesScreen({route, navigation}: ScreenProps) {
+  const address = route.params.address;
+  const modalRef = useRef<Modalize>(null);
+  const {data: identityInfo} = useAccountIdentityInfo(address);
+  const {data: subIdentitiesData} = useSubIdentities(address);
+  const [subIdentities, setSubIdentities] = useState(subIdentitiesData || []);
+  const [submitSubsDisabled, setSubmitSubsDisabled] = useState(true);
+  const queryClient = useQueryClient();
+  const startTx = useApiTx();
+
+  useEffect(() => {
+    if (subIdentitiesData?.length) {
+      setSubIdentities(subIdentitiesData);
+    }
+  }, [subIdentitiesData, setSubIdentities]);
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerRight: () => (
+        <TopNavigationAction
+          icon={(props) => <Icon {...props} name="plus-circle-outline" />}
+          onPress={onAddSubIdentityPress}
+        />
+      ),
+    });
+  });
+
+  const onAddSubIdentityPress = () => {
+    modalRef.current?.open();
+  };
+
+  const onSetSubIdentitiesPress = async () => {
+    setSubmitSubsDisabled(true);
+    startTx({
+      address,
+      txMethod: 'identity.setSubs',
+      params: [subIdentities.map((sub) => [sub.accountId, {raw: sub.name}])],
+    })
+      .then(() => {
+        queryClient.invalidateQueries(['sub-identities', address]);
```

Comment:
I think the query key is different in the hook, how about
```suggestion
        queryClient.invalidateQueries(['sub-identities']);
```
it should clear everything starting with 'sub-identities'

Reply:
Yeah I forgot to update this one after updating the queryKey in the hook.. I think `['sub-identities', {address}]` would also work to invalidate only the cache for the specific address